### PR TITLE
Support the latest pkg:xml, misc lint and language cleanup

### PR DESCRIPTION
**Pull request by kevmoo**
Prepare for v2.1.2

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -23,50 +23,34 @@ analyzer:
 
 linter:
   rules:
-    - depend_on_referenced_packages
```
Comment (kevmoo) [inline analysis_options.yaml:26]: Eliminated a LOT of duplicates with values in lints/recommended

**Comment by kevmoo** [comment]
Thanks! Could we publish? https://pub.dev/packages/stagexl

**Comment by AndrewLugg** [comment]
Yeah, I will publish later today.

**Comment by AndrewLugg** [comment]
Changes have been published now.

**Pull request merged**